charmhelpers/contrib/openstack/files/check_openvswitch.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# -*- coding: us-ascii -*-
+"""Check for OVS interface errors."""
+
+import argparse
+import json
+import subprocess
+import sys
+
+# dependence on nagios_plugin3 being installed by charm-nrpe
+from nagios_plugin3 import CriticalError, UnknownError, try_check
+
+
+def parse_ovs_status():
+    """Check for errors in 'ovs-vsctl show' output."""
+    try:
+        cmd = ["/usr/bin/sudo", "/usr/bin/ovs-vsctl", "--format=json", "list", "Interface"]
+        ovs_output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        raise UnknownError(
+            "UNKNOWN: Failed to query ovs: {}".format(
```

Comment:
Being more explicit about what failed, would be nice, e.g. s/Failed to query ovs/ovs-vsctl list Interface failed/ as this makes it less likely the Operator has to dig up the charm code to figure out what failed?